src/lib.rs

```diff
@@ -662,12 +667,35 @@ impl<T: Clone> Sender<T> {
     /// assert_eq!(s.broadcast(2).await, Err(SendError(2)));
     /// # });
     /// ```
-    pub fn broadcast(&self, msg: T) -> Send<'_, T> {
-        Send {
+    pub fn broadcast(&self, msg: T) -> Pin<Box<Send<'_, T>>> {
+        Box::pin(self.broadcast_direct(msg))
+    }
+
+    /// Broadcasts a message on the channel without pinning the future to the heap.
+    ///
+    /// The future returned by this method is not `Unpin` and must be pinned before use. This is
+    /// the desired behavior if you just `.await` on the future. For other uses cases, use the
```

Comment:
I keep forgetting so remind me: if one .awaits an unpinned future in an async block/function, does the async block/function also becomes unpinned? If so, "just" misrepresents the reality. If not, all good. :)

Reply:
`await`ing the future implicitly pins it to the stack.

Reply:
right, thanks. I don't know why but I can't keep all the pinning rules in head. :)